crystal_diffusion/models/mtp.py

```diff
@@ -0,0 +1,315 @@
+from typing import Any, Dict, List, Optional, Tuple
+
+import numpy as np
+import pandas as pd
+from maml.apps.pes import MTPotential
+from maml.utils import check_structures_forces_stresses, pool_from
+from monty.io import zopen
+from monty.tempfile import ScratchDir
+from pymatgen.core import Structure
+
+
+class MTPWithMLIP3(MTPotential):
+    """MTP with MLIP-3."""
+
+    def __init__(self,
+                 mlip_path: str,
+                 name: Optional[str] = None,
+                 param: Optional[Dict[Any, Any]] = None,
+                 version: Optional[str] = None):
+        """Modifications to maml.apps.pes._mtp.MTPotential to be compatible with mlip-3.
+
+        Args:
+            mlip_path: path to mlip3 library
+            name: MTPotential argument
+            param: MTPotential argument
+            version: MTPotential argument
+        """
+        super().__init__(name, param, version)
+        self.mlp_command = os.path.join(mlip_path, "build", "mlp")
+        assert os.path.exists(self.mlp_command), "mlp command not found in mlip-3 build folder"
+        self.mlp_templates = os.path.join(mlip_path, "MTP_templates")
+        assert os.path.exists(self.mlp_templates), "MTP templates not found in mlip-3 folder"
+        self.fitted_mtp = None
+        self.elements = None
+
+    def to_lammps_format(self):
+        """Write the trained MTP in a LAMMPS compatible format."""
+        # TODO
+        # write params write the fitted mtp in a LAMMPS compatible format
+        # self.write_param(
+        #    fitted_mtp=fitted_mtp,
+        #    Abinitio=0,
+        #    Driver=1,
+        #    Write_cfgs=predict_file,
+        #    Database_filename=original_file,
+        #    **kwargs,
+        # )
+        pass
+
+    def evaluate(self,
+                 test_structures: List[Structure],
+                 test_energies: List[float],
+                 test_forces: List[List[float]],
+                 test_stresses: Optional[List[List[float]]] = None,
+                 ) -> Tuple[pd.DataFrame, pd.DataFrame]:
+        """Evaluate energies, forces, stresses and MaxVol gamma factor of structures with trained MTP.
+
+        Args:
+            test_structures: evaluation set of pymatgen Structure Objects.
+            test_energies: list of total energies of each structure to evaluation in test_structures list.
+            test_forces: list of calculated (m, 3) forces of each evaluation structure with m atoms in structures list.
+                m can be varied with each single structure case.
+            test_stresses (optional): list of calculated (6, ) virial stresses of each evaluation structure in
+                test_structures list. If None, do not evaluate on stresses. Default to None.
+
+        Returns:
+            dataframe with ground truth energies, forces
+            dataframe with predicted energies, forces, MaxVol gamma (nbh grades)
+        """
+        original_file = "original.cfgs"
```

Comment:
Here I would check that `self.fitted_mtp` is not `None` and return an error message telling the user that a model must be trained before it can be evaluated. 

I suspect that if we just let the Popen crash, the error message will be cryptic.

Reply:
good idea, Added a check with raise error